cmake/sdlchecks.cmake

```diff
@@ -1,17 +1,25 @@
 include(CMakeParseArguments)
+include(${SDL2_SOURCE_DIR}/cmake/sdlfind.cmake)
 macro(FindLibraryAndSONAME _LIB)
   cmake_parse_arguments(FLAS "" "" "LIBDIRS" ${ARGN})
 
   string(TOUPPER ${_LIB} _UPPERLNAME)
   string(REGEX REPLACE "\\-" "_" _LNAME "${_UPPERLNAME}")
 
   find_library(${_LNAME}_LIB ${_LIB} PATHS ${FLAS_LIBDIRS})
+  get_filename_component(${_LNAME}_EXT ${${_LNAME}_LIB} EXT)
+  if(${_LNAME}_EXT STREQUAL CMAKE_STATIC_LIBRARY_SUFFIX)
+      set(${_LNAME}_SHARED FALSE)
+  else()
+      set(${_LNAME}_SHARED TRUE)
+  endif()
   if(${_LNAME}_LIB)
```

Comment:
I think we need to be careful here, because on Linux the shared library can be manually set to something like `libasound.so.2`.
Perhaps it should be a regex match and also test for shared libraries.
On the Windows platform, you have to consider static import libraries.
Then it becomes:
```cmake
if(${_LNAME}_LIB MATCHES ".*${CMAKE_SHARED_LIBRARY_SUFFIX}.*" AND NOT ${_LNAME}_LIB MATCHES ".*${CMAKE_STATIC_LIBRARY_SUFFIX}.*")
    set(${_LNAME}_SHARED TRUE)
else()
    set(${_LNAME}_SHARED FALSE)
endif()
```
```